src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -409,6 +410,33 @@ public static void ResetCache()
             }
         }
 
+        private static void DeleteRecursivelyWithAttempts(string destinationDir)
+        {
+            const int magic = 10;
+            for (var elves = 1; elves <= magic; elves++)
+            {
+                try
+                {
+                    Directory.Delete(destinationDir, true);
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return;  // good!
+                }
+                catch (IOException)
+                { // System.IO.IOException: The directory is not empty
+                    Thread.Sleep(10);
+                    continue;
+                }
+                catch (UnauthorizedAccessException)
```

Comment:
I'm a bit concerned about catching `UnauthorizedAccessException`. Deleting such directory requires elevated permission. This type of exception is related to user's directory permission setting, and should be surfaced up to user instead of silently eating it at the backend.

Reply:
This directory was created by us to add the rules files in, we should be able to delete it permission wise if we created it. This is more to catch conflicting access issues and wait for those to clear up which can happen sometimes if we just read from that directory for example. Making sure all handles are released properly which can fail if we only try once.